Engineering/Frontend/Shadow Dom.md

```diff
@@ -0,0 +1,38 @@
+---
+tags: frontend, shadow dom, engineering/frontend
+author: Tran Khac Vy
+github_id: trankhacvy
+date: 2023-05-16
+---
+
+Recently, I have endeavored to develop an application on the platform of chat GPT, bearing the name Javis. It functions as a chrome extension, serving as an AI assistant to aid users during their web browsing experience. Upon testing the application on Chrome, I encountered an issue where Javis's CSS conflicted with the CSS of the running webpage, resulting in both my application and the webpage occasionally displaying inaccurately compared to their original interfaces. After conducting an extensive online search, I discovered a viable solution known as shadow dom. In this article, I shall introduce you to the concept of shadow dom and elucidate the myriad benefits it bestows.
```

Comment:
The introduction sounds a bit hard to follow the objective of the post. I would suggest we have a clearer version:

> My latest venture involves the development of an application called Javis, which operates as a Chrome extension. Functioning as an AI assistant, Javis is designed to facilitate a smooth and enriching web browsing experience for users. During its testing phase in Chrome, I faced a unique challenge - **Javis's CSS clashed with the webpage's CSS**. The result was a distortion of both the application's and the webpage's interfaces. Following an extensive research, I unearthed a potent solution - the Shadow DOM. In this piece, I aim to explain the concept of Shadow DOM and shed light on the numerous advantages it offers.